modules/SDL.lua

```diff
@@ -33,33 +33,29 @@ function SDL:StartSDL(pathToSDL, smartDeviceLinkCore, ExitOnCrash)
   end
   local status = self:CheckStatusSDL()
 
-  while status == self.RUNNING do
-    sleep(1)
-    print('Waiting for SDL shutdown')
-    status = self:CheckStatusSDL()
+  if (status == self.RUNNING) then
+    local msg = "SDL had already started out of ATF"
+    xmlReporter.AddMessage("StartSDL", {["message"] = msg})
+    print(console.setattr(msg, "cyan", 1))
+    return nil, msg
   end
 
-  if status == self.STOPPED  or status == self.CRASH then
-    CopyInterface()
-    local result = os.execute ('./tools/StartSDL.sh ' .. pathToSDL .. ' ' .. smartDeviceLinkCore)
-    if config.storeFullSDLLogs == true then
-      sdl_logger.init_log(get_script_file_name())
-    end
-    if result then
-      local msg = "SDL started"
-      xmlReporter.AddMessage("StartSDL", {["message"] = msg})
-      return true
-    else
-      local msg = "SDL had already started not from ATF or unexpectedly crashed"
-      xmlReporter.AddMessage("StartSDL", {["message"] = msg})
-      print(console.setattr(msg, "cyan", 1))
-      return nil, msg
-    end
+  CopyInterface()
+  local result = os.execute ('./tools/StartSDL.sh ' .. pathToSDL .. ' ' .. smartDeviceLinkCore)
+  if config.storeFullSDLLogs == true then
+    sdl_logger.init_log(get_script_file_name())
   end
-  local msg = "SDL had already started from ATF"
-  xmlReporter.AddMessage("StartSDL", {["message"] = msg})
-  print(console.setattr(msg, "cyan", 1))
-  return nil, msg
+  if result then
+    local msg = "SDL started"
+    xmlReporter.AddMessage("StartSDL", {["message"] = msg})
+    return true
+  else
+    local msg = "SDL had already started not from ATF or unexpectedly crashed"
+    xmlReporter.AddMessage("StartSDL", {["message"] = msg})
+    print(console.setattr(msg, "cyan", 1))
+    return nil, msg
```

Comment:
@VVeremjova please avoid returning different count of args and different types of the, If you return bool as furst parameter - use, return false in other case. Also on success retsult you can return message "SDL started successfully" 
Also to avoid duplicating same xmlRporting and avoid putting logs initialisation between start SDL, and checking that is is started, you is able just initialise variables inside if, and move return below. 
Also I suppose, that you should init sdl_los only in case if SDL was started.